### PR TITLE
Cover major themes in Azure 11.2.1 release notes

### PR DESCRIPTION
**Pull request by sslavic**
@MartaD you defined the empty release notes file with title only and committed it directly to master (no PR, review) I don't get it why https://github.com/giantswarm/releases/commit/d38b51a5a3a1641fdf0a5fe547729cf86a2471b7

Anyway, this PR is extraction of Azure 11.2.1 release notes from https://github.com/giantswarm/releases/pull/141 (will remove it there)
It's building on top of changes done on master, instead of resolving conflicts in the original PR.

### CODE REVIEW

File: release-notes/azure/v11.2.1.md [hunk 1]
```diff
@@ -1 +1,53 @@
 ## :zap: Giant Swarm Release 11.2.1 for Azure is now active for you! :zap:
+
+**With this release, horizontal pod autoscaling (HPA) is enabled by default for NGINX Ingress Controller (NGINX IC) on selected cluster profiles.**
+
+Here, you will find the NGINX IC to be more elastic thus driving down your costs.
```
Comment (cokiengchiara) [inline release-notes/azure/v11.2.1.md]:
```suggestion
Here, you will find the NGINX IC to be more elastic thus driving down your costs. The criteria for which cluster profiles are selected will be adjusted as we collect data from running in production.
```

File: release-notes/azure/v11.2.1.md [hunk 2]
```diff
@@ -1 +1,53 @@
...
+
+Version development included work with the upstream maintainer. Prior to release, we ran trials with customers. We performed these tests on small and large clusters alike, in order to ensure stability and scalability.
+
+This version also includes an upgrade to the latest upstream NGINX IC 0.30.0, as well as improvements to other components (cluster-operator, kube-state-metrics, chart-operator) as detailed in the changelog.
```
Comment (cokiengchiara) [inline release-notes/azure/v11.2.1.md]: @sslavic can you pick top 3 most important component changes for Azure? Seems azure-operator and k8s upgrade might be more important than cluster-operator, chart-operator, etc.?

File: release-notes/azure/v11.2.1.md (hunk 1 above)
Comment (sslavic) [reply release-notes/azure/v11.2.1.md]: synced

File: release-notes/azure/v11.2.1.md (hunk 2 above)
Comment (sslavic) [reply release-notes/azure/v11.2.1.md]: Done, PTAL

**Pull request merged**